src/mlpack/core/data/pre_processing_impl.hpp

```diff
@@ -0,0 +1,136 @@
+#ifndef MLPACK_CORE_DATA_DATA_PRE_PROCESSING_IMPL_HPP
+#define MLPACK_CORE_DATA_DATA_PRE_PROCESSING_IMPL_HPP
+
+#include "pre_processing.hpp"
+
+namespace mlpack {
+namespace data {
+  Data::Data(std::string file_path)
+  {
+    // Constructor
+    path = file_path;
+    mlpack::data::Load(path, data, true);
+    data = data.submat(0, 1, data.n_rows - 1, data.n_cols - 1);
+    getHeaders();
+    getNumericAttributes();
+  }
+
+  void Data:: info()
+  {
+    // Returns Index range, list of headers
+    std::cout << typeid(data).name() << std::endl;
+    std::cout << "Index Range: " << data.n_cols << std::endl;
+    std::cout << "Numeric Attributes: " << numericData.n_rows << std::endl;
+    std::cout << "Non-numeric Attributes: " << headers.size() - numericData.n_rows << std::endl;
+    std::cout << "Data Columns (total " << data.n_rows << "):" << std::endl;
```

Comment:
Let's call this `dimensions` not `columns`.  Since we represent points as columns in C++, but users may be used to representing points as rows, it's best to try and use language agnostic to the data representation. :+1: